lib/ansible/modules/messaging/rabbitmq/rabbitmq_user.py

```diff
@@ -211,28 +212,28 @@ def set_tags(self):
         self._exec(['set_user_tags', self.username] + self.tags)
 
     def set_permissions(self):
-        for permission in self._permissions:
-            if permission not in self.permissions:
-                cmd = ['clear_permissions', '-p']
-                cmd.append(permission['vhost'])
-                cmd.append(self.username)
-                self._exec(cmd)
-        for permission in self.permissions:
-            if permission not in self._permissions:
-                cmd = ['set_permissions', '-p']
-                cmd.append(permission['vhost'])
-                cmd.append(self.username)
-                cmd.append(permission['configure_priv'])
-                cmd.append(permission['write_priv'])
-                cmd.append(permission['read_priv'])
-                self._exec(cmd)
+        permissions_to_clear = [permission for permission in self._permissions if permission not in self.permissions]
+        permissions_to_add = [permission for permission in self.permissions if permission not in self._permissions]
```

Comment:
The only nit-pick I can find here is that `self._permissions` and `self.permissions` looks very similar. To increase readability you could var them separately
```suggestion
        current_perms = set(self._permissions)
        new_perms = set(self.permissions)
        permissions_to_clear = current_perms - new_perms  # using set.difference(), or the list comprehension you used
        permissions_to_add = new_perms - current_perms  # using set.difference(), or the list comprehension you used
```

This makes it more clear that you are trying to check what to add/remove based on what permissions are currently there vs what is requested.